src/Host/Client/Impl/Host/LocalBrokerClient.cs

```diff
@@ -24,14 +24,15 @@ internal sealed class LocalBrokerClient : BrokerClient {
         private const string InterpreterId = "local";
 
         private static readonly bool ShowConsole;
-        private static readonly LocalCredentialsDecorator Credentials = new LocalCredentialsDecorator();
+        private static readonly LocalCredentialsDecorator _credentials = new LocalCredentialsDecorator();
 
         private readonly string _rhostDirectory;
         private readonly string _rHome;
         private readonly BinaryAsyncLock _connectLock = new BinaryAsyncLock();
         private readonly ICoreServices _services;
 
         private Process _brokerProcess;
+        private int _disposed;
```

Comment:
If I remember correctly, `DisposableBag` implements this for you. @AlexanderSher ?